asmcomp/selectgen.ml

```diff
@@ -194,18 +195,139 @@ let join_array rs =
 (* Name of function being compiled *)
 let current_function_name = ref ""
 
+(* Environment parameter for the function being compiled, if any. *)
+let current_function_env_param = ref None
```

Comment:
(Actually I've incorporated the old patch of mine, but not yet the one which propagates mutability information from the middle-end.  I'm going to let @lpw25 submit that separately.)